linkerd/admin/src/main/resources/io/buoyant/linkerd/admin/js/router_server.js

```diff
@@ -1,28 +1,106 @@
 var RouterServer = (function() {
-  // Skeletal implementation to demonstrate functionality of RouterController
+  var template;
 
-  function initializeServerContainers(servers, $serverEl) {
-    _.each(servers, function(server) {
-      var $se = $("<div />").addClass("router-server")
-        .html("<div>Servers</div><div>" + server.prefix + "</div>");
-      $se.appendTo($serverEl);
+  function renderServer($container, server, data) {
+    var serverHtml = template({
+      server: server.label,
+      metrics: data
     });
-  }
+    var $serverHtml = ($("<div />").addClass("router-server").html(serverHtml));
 
-  function renderServers() {
+    $container.html($serverHtml);
   }
 
-  return function (routers, $serverEl, routerName) {
-    var servers = routers.servers(routerName);
-    initializeServerContainers(servers, $serverEl);
+  function processData(data, metricDefinitions) {
+    var lookup = {}; // track # requests for SuccessRate()
 
-    return {
-      onMetricsUpdate: function() {
-        renderServers();
-      },
-      desiredMetrics: function () {
-        return [];
+    return _.map(metricDefinitions, function(defn) {
+      var serverData = Query.filter(defn.query, data);
+
+      if (!_.isEmpty(serverData)) {
+        defn.value = serverData[0].delta;
+        lookup[defn.metricSuffix] = defn.value;
+
+        if(_.isFunction(defn.getRate)) {
+          defn.rate = defn.getRate(lookup);
+        }
       }
+
+      return defn;
+    });
+  }
+
+  return function (metricsCollector, server, $serverEl, routerName, serverTemplate, metricDefinitions) {
+    template = serverTemplate;
+
+    var metricsHandler = function(data) {
+      var filteredData = _.filter(data.specific, function (d) { return d.name.indexOf(routerName) !== -1 });
+      var transformedData = processData(filteredData, metricDefinitions);
+      renderServer($serverEl, server, transformedData);
+    }
+
+    var getDesiredMetrics = function(metrics) {
+      return _.map(metricDefinitions, function(d) {
+        return Query.filter(d.query, metrics);
+      });
     }
+
+    metricsCollector.registerListener(metricsHandler, getDesiredMetrics);
+
+    return {};
   };
 })();
+
+var RouterServers = (function() {
+  function getMetricDefinition(routerName, serverName) {
```

Comment:
tioli, this method would probably work just as well inside `RouterServer` where it's actually used